File: reactive-spring-cloud-starter/src/main/java/com/jmsoftware/maf/reactivespringcloudstarter/controller/CommonController.java
package com.jmsoftware.maf.reactivespringcloudstarter.controller;

import cn.hutool.json.JSON;
import com.jmsoftware.maf.common.bean.ResponseBodyBean;
import com.jmsoftware.maf.common.domain.ValidationTestPayload;
import com.jmsoftware.maf.reactivespringcloudstarter.service.CommonService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.*;

/**
 * <h1>CommonController</h1>
 * <p>
 * Change description here.
 *
 * @author Johnny Miller (锺俊), email: johnnysviva@outlook.com
 * @date 2/4/20 10:29 AM
 **/
@Slf4j
@RestController
@RequiredArgsConstructor
@RequestMapping("/common")
public class CommonController {
    private final CommonService commonService;

    @GetMapping("/app-info")
    public ResponseBodyBean<JSON> applicationInformation() {
        return ResponseBodyBean.ofSuccess(this.commonService.getApplicationInfo(), "Succeed to retrieve app info.");
    }

    @PostMapping("/validation-test")
    public ResponseBodyBean<String> validationTest(@RequestBody ValidationTestPayload payload) {
        this.commonService.validateObject(payload);
        return ResponseBodyBean.ofSuccess(payload.getName(), "validationTest()");
    }
}

File: reactive-spring-cloud-starter/src/main/kotlin/com/jmsoftware/maf/reactivespringcloudstarter/controller/CommonController.kt
package com.jmsoftware.maf.reactivespringcloudstarter.controller

import cn.hutool.json.JSON
import com.jmsoftware.maf.common.bean.ResponseBodyBean
import com.jmsoftware.maf.common.domain.ValidationTestPayload
import com.jmsoftware.maf.reactivespringcloudstarter.service.CommonService
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.*
import javax.validation.Valid

/**
 * # CommonController
 *
 * Change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/16/22 11:37 AM
 */
@Validated
@RestController
@RequestMapping("/common")
class CommonController(
    private val commonService: CommonService
) {
    @GetMapping("/app-info")
    fun applicationInformation(): ResponseBodyBean<JSON> {
        return ResponseBodyBean.ofSuccess(commonService.getApplicationInfo(), "Succeed to retrieve app info.")
    }

    @PostMapping("/validation-test")
    fun validationTest(@Valid @RequestBody payload: ValidationTestPayload): ResponseBodyBean<String> {
        commonService.validateObject(payload)
        return ResponseBodyBean.ofSuccess(payload.name, "validationTest()")
    }
}
